.github/workflows/prepare_release.yml

```diff
@@ -0,0 +1,67 @@
+---
+
+name: Prepare Release
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        description: 'Module version to be released.'
+        required: true
+        type: string
+      allowed_owner:
+        description: 'Only allow this owner'
+        required: true
+        type: string
+      working-directory:
+        description: The working directory where all jobs should be executed. Used for modules in subdirectories like a monorepo or a control repository.
+        default: '.'
+        required: false
+        type: string
+    secrets:
+      username:
+        description: 'The forge username to use'
+        required: true
+
+env:
+  BUNDLE_WITHOUT: development:test:system_tests
+
+jobs:
+  prepare_release:
+    defaults:
+      run:
+        working-directory: ${{ inputs.working-directory }}
+    name: 'Puppet Forge'
+    runs-on: ubuntu-24.04
+    if: github.repository_owner == inputs.allowed_owner
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.3'
+          bundler-cache: true
+          working-directory: ${{ inputs.working-directory }}
+      - name: Update metadata.json to new version
+        run: bundle exec rake module:bump_to_version[${{ inputs.version }}]
+      - name: Update CHANGELOG.md and update REFERENCE.md if required
+        env:
+          # https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication
+          CHANGELOG_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: bundle exec rake release:prepare
+      - name: 'Check if a release is necessary'
+        id: 'check'
+        run: |
+          git diff --quiet CHANGELOG.md && echo "release=false" >> $GITHUB_OUTPUT || echo "release=true" >> $GITHUB_OUTPUT
+      - name: 'Commit changes'
+        if: ${{ steps.check.outputs.release == 'true' }}
+        env:
+          # https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config --local user.email "${{ github.repository_owner }}@users.noreply.github.com"
+          git config --local user.name "GitHub Actions"
+          git add .
+          git commit -m "Release v${{ steps.get_version.outputs.version }}"
+          gh pr create --title "Release v${{ steps.get_version.outputs.version }}" --label skip-changelog
```

Comment:
In Foreman we use [peter-evans/create-pull-request](https://github.com/peter-evans/create-pull-request) (see https://github.com/theforeman/foreman-packaging/blob/605edf36c9cad3e062ce855601e52b93442c8fa8/.github/workflows/bump_packages.yml#L105-L112) and that has the nice benefit that it will update an existing PR by reusing a branch.

Reply:
The current workflow doesn't switch to another branch and probably commits straight to master.

Reply:
It creates a `release_prep` branch now. Perforce uses the same branch name.

Reply:
Perforce doesn't maintain stable branches so they don't need it. If they did, we wouldn't have the whole stdlib 9.x debacle and could have easily created an 8.x release with the non-deprecated functions.

Reply:
I know. But their release branch is always named `release_prep`, I simply wanted to copy that.